src/Symfony/Component/Yaml/Inline.php

```diff
@@ -506,7 +506,12 @@ private static function parseMapping($mapping, $flags, &$i = 0, $references = []
 
             if ('!php/const' === $key) {
                 $key .= self::parseScalar($mapping, $flags, [':', ' '], $i, false, [], true);
-                $key = self::evaluateScalar($key, $flags);
+                if ('!php/const:' === $key && ':' !== $mapping[$i]) {
+                    $key = '';
```

Comment:
No need to evaluate because we know it's gonna end up as the empty string (the key === `php/const`).